file/watch.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+	"path/filepath"
+)
+
+func Watch(filename string, callback func()) error {
```

Comment:
I am wondering if it is interesting to provide it as a task.
Like that you will have a context management behind the seen.

Reply:
i dont understand the interest to do this
do you have a use case example ?

Reply:
implementing a task as the way it is defined in the package `async`, will force you to handle a context. Context that will be used by the package `app` to smoothly stopped any kind go routine created as long as it is a tasked.

But as suggested by @celian-garcia, the package could propose a function and a task so you can choose the way you want to watch a file. I can understand that implementing a task can be heavy, so maybe proposing both approach can be interesting.